lib/interface.js

```diff
@@ -81,31 +81,17 @@ module.exports = function(Target) {
     );
   };
 
-  // Creates a method which "coerces" to a promise, by calling a
-  // "then" method on the current `Target`
-  each(
-    [
-      'bind',
-      'catch',
-      'finally',
-      'asCallback',
-      'spread',
-      'map',
-      'reduce',
-      'thenReturn',
-      'return',
-      'yield',
-      'ensure',
-      'reflect',
-      'get',
-      'mapSeries',
-      'delay',
-    ],
-    function(method) {
-      Target.prototype[method] = function() {
-        const promise = this.then();
-        return promise[method].apply(promise, arguments);
-      };
-    }
-  );
+  Target.prototype.asCallback = function(cb) {
+    const promise = this.then();
+    callbackify(() => promise)(cb);
+    return promise;
+  };
+
+  Target.prototype.catch = function(onReject) {
+    return this.then().catch(onReject);
+  };
+
+  Target.prototype.finally = function(onFinally) {
+    return this.then().finally(onFinally);
+  };
```

Comment:
@maximelkin : Based upon our earlier discussion, I think this one should probably be conditionally injected as well.